### ISSUE TITLE
Selection problem (RTL and LTR)

### ISSUE THREAD

**Issue by abdullahO2** [notepad-plus-plus/notepad-plus-plus#11658]
### Description of the Issue
There is a problem in defining Arabic words when the text direction is from left to right, and the opposite happens when converting the page direction. English words face the same problem

### Steps to Reproduce the Issue
1. Put a text consisting of sentences in Arabic and English such as: "Initial مرحبا أحمد the loading screen"
2. Try to select Arabic words when the page orientation is from left to right.
3. Try to identify the English words when the direction of the page is from right to left.

### Expected Behavior
The words are selected correctly

### Actual Behavior
The wrong word is selected

![image](https://i.ibb.co/NVMXJRt/notepad-ii-E6-YFL6-E9.gif)

### Debug Information
Notepad++ v8.4.1   (64-bit)
Build time : May  8 2022 - 19:51:18
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Users\kk\Downloads\app_en.arb"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 (64-bit) 
OS Version : 2009
OS Build : 22000.675
Current ANSI codepage : 1256
Plugins : ComparePlugin.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by rdipardo** [comment]
Known issue, (from even before the Scintilla upgrade in v8.4): <https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11476#issuecomment-1094276395>

> Double-clicking a Hebrew word in **RTL**, the word is selected properly. Double-clicking a Hebrew word in **LTR**, random characters are selected.
>

**Comment by Kalbod** [comment]
When there is Persian and English text in one line, you can't edit Persian (or Arabic) and English text at the same time. You have to right-align (RTL) and edit Persian, and left-align(LTR) and edit English, and that's ridiculous.

**Comment by xomx** [comment]
@Kalbod 
Yes, N++ is not currently a suitable editor for that ([BiDi](https://en.wikipedia.org/wiki/Bidirectional_text)).
For more details see e.g. here: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10664
